### PR TITLE
Added build-retention scripts.

### PR DESCRIPTION
**Pull request by nitisdev**
Signed-off-by: Nitish Singh <nitish.singh@seagate.com>

# Problem Statement
- Added build-retention scripts.

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design
1. Added build-retention scripts.

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [x] Unit and System Tests are added
- [x] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [x] Testing was performed with RPM
- [x] Jenkins pipelines used for testing

https://eos-jenkins.colo.seagate.com/job/Automation/job/Build%20Retention/
http://10.230.242.163:3000/d/xMkxip67z/lco-lab-nfs?orgId=1
http://10.230.242.163:9090/alerts?search=

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [x] Interface change (if any) are documented
- [x] Side effects on other features (deployment/upgrade)
- [x] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [x] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: scripts/build-retention/build_retention.groovy [hunk 1]
```diff
@@ -0,0 +1,52 @@
+pipeline {
```
Comment (shailesh-vaidya) [inline scripts/build-retention/build_retention.groovy:1]: Please move this file to jenkins/automation/build-retention

**Review by shailesh-vaidya — CHANGES_REQUESTED** [review]
Please add a document with steps to deploy Prometheus setup at /scripts/build-retention/monitoring

File: scripts/build-retention/build_retention.groovy (hunk 1 above)
Comment (nitisdev) [reply scripts/build-retention/build_retention.groovy]: Done

File: jenkins/automation/build_retention/build_retention.groovy [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+
+    }
+	
+	stages {
+		
+		stage('Checkout Script') {
+            steps {             
+                script {
+                    checkout([$class: 'GitSCM', branches: [[name: '*/Build-Retention']], doGenerateSubmoduleConfigurations: false, extensions: [], submoduleCfg: [], userRemoteConfigs: [[credentialsId: 'cortx-admin-github', url: 'https://github.com/nitisdev/cortx-re']]])
```
Comment (gauravchaudhari02) [inline jenkins/automation/build_retention/build_retention.groovy:22]: Change repo and branch to main once testing is done

File: scripts/build-retention/monitoring/prometheus.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+rule_files:
+# All mountpoints in alert.rules have been manually mounted onto the node.
+  - 'alert.rules'
+
+# Change all ips below to that of the mounted node.
+scrape_configs:
+  - job_name: 'prometheus'
+    static_configs: 
+      - targets: ['10.230.242.163:9090']
```
Comment (gauravchaudhari02) [inline scripts/build-retention/monitoring/prometheus.yml:14]: Should not mention IP hardcoded in this file

File: scripts/build-retention/monitoring/alert.rules [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    for: 30s
+    labels:
+      severity: Critical
+    annotations:
+      summary: "/mnt/data1/releases storage is filling fast."
+      description: "/mnt/data1/releases usage is {{ humanize $value}}%. Reported by instance {{ $labels.instance }} of job {{ $labels.job }}."
+
+  - alert: jenkins-dev
+# Change all ips below to that of the mounted node.
+    expr: 100 - (100 * ((node_filesystem_avail_bytes{device="cortx-storage.colo.seagate.com:/mnt/data1/releases/jenkins/jenkins-dev/jenkins_home-dev", fstype="nfs4", instance="10.230.242.163:9100", job="node-exporter", mountpoint="/rootfs/mnt/jenkins-dev"} )  / (node_filesystem_size_bytes{device="cortx-storage.colo.seagate.com:/mnt/data1/releases/jenkins/jenkins-dev/jenkins_home-dev", fstype="nfs4", instance="10.230.242.163:9100", job="node-exporter", mountpoint="/rootfs/mnt/jenkins-dev"}) )) > 95
```
Comment (gauravchaudhari02) [inline scripts/build-retention/monitoring/alert.rules:26]: Should not mention IP hardcoded

File: scripts/build-retention/monitoring/alert.rules [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    for: 30s
+    labels:
+      severity: Critical
+    annotations:
+      summary: "/mnt/data1/releases/jenkins/jenkins-dev/jenkins_home-dev storage is filling fast."
+      description: "/mnt/data1/releases/jenkins/jenkins-dev/jenkins_home-dev usage is {{ humanize $value}}%. Reported by instance {{ $labels.instance }} of job {{ $labels.job }}."
+
+  - alert: jenkins-prod
+# Change all ips below to that of the mounted node.
+    expr: 100 - (100 * ((node_filesystem_avail_bytes{device="cortx-storage.colo.seagate.com:/mnt/data1/releases/jenkins/jenkins-prod", fstype="nfs4", instance="10.230.242.163:9100", job="node-exporter", mountpoint="/rootfs/mnt/jenkins-prod"} )  / (node_filesystem_size_bytes{device="cortx-storage.colo.seagate.com:/mnt/data1/releases/jenkins/jenkins-prod", fstype="nfs4", instance="10.230.242.163:9100", job="node-exporter", mountpoint="/rootfs/mnt/jenkins-prod"}) )) > 95
```
Comment (gauravchaudhari02) [inline scripts/build-retention/monitoring/alert.rules:36]: Should not mention IP hardcoded

File: scripts/build-retention/monitoring/alert.rules [hunk 6]
```diff
@@ -0,0 +1,42 @@
+groups:
+- name: alert.rules
+  rules:
+  - alert: cortx-storage
+# Change all ips below to that of the mounted node.
+    expr: 100 - (100 * ((node_filesystem_avail_bytes{device="cortx-storage.colo.seagate.com:/mnt/data1", fstype="nfs4", instance="10.230.242.163:9100", job="node-exporter", mountpoint="/rootfs/mnt/cortx-storage"} )  / (node_filesystem_size_bytes{device="cortx-storage.colo.seagate.com:/mnt/data1", fstype="nfs4", instance="10.230.242.163:9100", job="node-exporter", mountpoint="/rootfs/mnt/cortx-storage"}) )) > 95
```
Comment (gauravchaudhari02) [inline scripts/build-retention/monitoring/alert.rules:6]: please hardcoded IP's from this file

File: jenkins/automation/build_retention/build_retention.groovy (hunk 2 above)
Comment (nitisdev) [reply jenkins/automation/build_retention/build_retention.groovy]: Done

File: scripts/build-retention/monitoring/prometheus.yml (hunk 3 above)
Comment (nitisdev) [reply scripts/build-retention/monitoring/prometheus.yml:14]: IP is reference to build-retention node where deployment is running & LCO NFS is mounted. Developer will change IP on deploying to next node. Current IP shows deployed node and should be treated as placeholder reference.

File: scripts/build-retention/monitoring/alert.rules (hunk 4 above)
Comment (nitisdev) [reply scripts/build-retention/monitoring/alert.rules:26]: IP is reference to build-retention node where deployment is running & LCO NFS is mounted. Developer will change IP on deploying to next node. Current IP shows deployed node and should be treated as placeholder reference.

File: scripts/build-retention/monitoring/alert.rules (hunk 5 above)
Comment (nitisdev) [reply scripts/build-retention/monitoring/alert.rules:36]: IP is reference to build-retention node where deployment is running & LCO NFS is mounted. Developer will change IP on deploying to next node. Current IP shows deployed node and should be treated as placeholder reference.

File: scripts/build-retention/monitoring/alert.rules (hunk 6 above)
Comment (nitisdev) [reply scripts/build-retention/monitoring/alert.rules:6]: IP is reference to build-retention node where deployment is running & LCO NFS is mounted. Developer will change IP on deploying to next node. Current IP shows deployed node and should be treated as placeholder reference.

**Review by gauravchaudhari02 — DISMISSED** [review]

**Comment by nitisdev** [comment]
> Please add a document with steps to deploy Prometheus setup at /scripts/build-retention/monitoring

Added README.md with screenshots.

**Pull request closed without merging**